src/main/java/org/example/DiagonalMatrix.java

```diff
@@ -2,83 +2,30 @@
 
 import java.util.Scanner;
 
-import static java.lang.Math.abs;
+import static org.example.DoWhileLoopMatrixGenerator.generateMatrixWithDoWhileLoop;
+import static org.example.ForLoopMatrixGenerator.generateMatrixWithForLoop;
+import static org.example.WhileLoopMatrixGenerator.generateMatrixWithWhileLoop;
 
 public class DiagonalMatrix {
     public static void main(String[] args) {
-        // Create a Scanner object to read input from the console
         Scanner scanner = new Scanner(System.in);
-
-        // Prompt the user to enter the value of 'n'
         System.out.print("Enter the value of 'n' (not greater than 100): ");
         int n = scanner.nextInt();
-                // Check if n is greater than 100
+
         if (n > 100) {
             System.out.println("Please enter a value less than or equal to 100.");
             return;
         }
-        int[][] matrix = new int[n][n];  // initialization of the matrix
-        int i=0;  // this variable will be used to parse the 2D matrix by row
-        int j=0;  // this variable will be used to parse the 2D matrix by column
 
-        System.out.println("Example with while:");
-        // generating and filling the matrix
-        while (i<n){
-            while (j<n) {
-                matrix[i][j]=abs(i-j);   //using abs predifined function to return the absolute value.
-                j++;
-            }
-            i++;
-            j=0;  //setting the j back to 0 for the next row
-        }
-        i=0;  // setting the i back to 0 for the next While
-        // Print the generated matrix
-        while (i<n){
-            while (j<n){
-                System.out.print(matrix[i][j]+ " ");
-                j++;
-            }
-            i++;
-            j=0;
-            System.out.println();
-        }
-        i=0;
+        int[][] matrix = new int[n][n];
```

Comment:
who is responsible for creating the matrix, `main` method or `generateMatrixWith...(matrix, n)` methods?

Reply:
at this point, each method from the loops is responsible for creating and printing the loops. So I have moved the initialization of the matrix inside the loop methods and sent as a parameter only the n number